### ISSUE TITLE
UUID, pathlib.Path not unstructured

### ISSUE THREAD

**Issue by ofekd** [python-attrs/cattrs#338]
* cattrs version: 22.2.0
* Python version: 3.10.9
* Operating System: Arch Linux 6.1.2

### Description

Some types are not unstructured properly -- I hope I'm not missing something obvious here, tried simply executing a sample from the docs.

### What I Did

```
[project]
name = "test"
version = "0.1.0"
description = "test"
dependencies = [
    "cattrs==22.2.0",
]
requires-python = ">=3.10,<3.12"
license = {text = "GPL-3.0-or-later"}
readme = "README.md"
```

```
$ pdm install
$ pdm run python
Python 3.10.9 (main, Dec 19 2022, 17:35:49) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> import cattrs
>>> from uuid import uuid4
>>> cattrs.unstructure(Path("."))
PosixPath('.')
>>> cattrs.unstructure(uuid4())
UUID('4ec32dab-894a-4552-908f-8ea3684598a9')
```

**Comment by Tinche** [comment]
Path support hasn't been released yet (soon, in 23.1.0), and UUID I don't think we support yet (we could).

**Comment by ofekd** [comment]
Got it. Didn't notice I was seeing docs for a newer version. Thank you
